src/hooks/useIntersectionObserver.js

```diff
@@ -1,67 +1,79 @@
-                        if (id === activeCategory) {
-                            dispatch({
-                                type: actionTypes.ACTIVE_CATEGORY_SET,
-                                activeCategory: null
-                            });
-                        } else if (id === GROUP_NAME_RECENTLY_USED) {
-                            const nextSibling = target.nextSibling;
-
-                            if (nextSibling) {
-                                const id = nextSibling.getAttribute(PROPERTY_DATA_NAME);
+            if (entry.intersectionRatio === 0) {
+              if (id === activeCategory) {
+                dispatch({
+                  type: actionTypes.ACTIVE_CATEGORY_SET,
+                  activeCategory: null,
+                });
+              } else if (id === GROUP_NAME_RECENTLY_USED) {
+                const nextSibling = target.nextSibling;
 
-                                dispatch({
-                                    type: actionTypes.ACTIVE_CATEGORY_SET,
-                                    activeCategory: id
-                                });
-                            }
-                        }
+                if (nextSibling) {
+                  const id = nextSibling.getAttribute(PROPERTY_DATA_NAME);
 
-                    } else if (!activeCategory) {
-                        dispatch({
-                            type: actionTypes.GROUP_SEEN_SET,
-                            group: id
-                        });
-                        dispatch({
-                            type: actionTypes.ACTIVE_CATEGORY_SET,
-                            activeCategory: id
-                        });
-                    }
-                });
-            }, {
-                root: root.current.parentElement
-            });
+                  dispatch({
+                    type: actionTypes.ACTIVE_CATEGORY_SET,
+                    activeCategory: id,
+                  });
+                }
+              }
+            } else if (!activeCategory) {
+              dispatch({
+                type: actionTypes.GROUP_SEEN_SET,
+                group: id,
+              });
+              dispatch({
+                type: actionTypes.ACTIVE_CATEGORY_SET,
+                activeCategory: id,
+              });
+            }
+          });
+        },
+        {
+          root: root.current.parentElement,
         }
+      );
+    }
 
-        observer.disconnect();
+    observer.disconnect();
 
-        if (!root || !root.current) {
-            return;
-        }
+    if (!root || !root.current) {
+      return;
+    }
 
-        [...root.current.querySelectorAll('.emoji-group')].forEach((target) => {
-            observer.observe(target);
-        });
+    [...root.current.querySelectorAll(".emoji-group")].forEach((target) => {
+      observer.observe(target);
+    });
 
-    }, [ root.current, filterResult ]);
+    return () => {
+      observer.disconnect();
```

Comment:
We weren't releasing the observers.

Reply:
It looks like we're releasing at line 54 (and now 63), though.

With this cleanup function we're covering the case in which the component was unmounted while the effect was still iterating on entries - but I don't think it would be that slow, or it would block the thread and the user would notice